pylabnet/hardware/nanopositioners/smaract.py

```diff
@@ -0,0 +1,370 @@
+""" Module for controlling Smaract open-loop nanopositioners using the MCS-2 controller """
+
+import ctypes
+
+from pylabnet.network.core.service_base import ServiceBase
+from pylabnet.utils.logging.logger import LogHandler
+from pylabnet.utils.helper_methods import value_to_bitval, bitval_to_value
+
+
+class MCS2:
+
+    # property keys and other constants
+    PKEY_NUM_BUS = int("0x020F0016", 16)
+    PKEY_NUM_CH = int('0x020F0017', 16)
+    PKEY_NUM_BUS_CH = int('0x02030017', 16)
+    PKEY_MOVE_MODE = int('0x03050087', 16)
+    PKEY_STEP_FREQ = int('0x0305002E', 16)
+    PKEY_STEP_AMP = int('0x03050030', 16)
+    PKEY_VOLT = int('0x0305001F', 16)
+    PKEY_DC_VEL = int('0x0305002A', 16)
+    PKEY_CURRENT_STATE = int('0x0305000F', 16)
+    MOVE_MODE_STEP = 4
+    MOVE_MODE_DC_REL = 3
+    MOVE_MODE_DC_ABS = 2
+    SCALE = 65535
+    DC_VEL_MIN = 100/56635
+    DC_VEL_MAX = 10**8
+
+    def __init__(self, logger=None):
+        """ Instantiate Nanopositioners"""
+
+        self.log = LogHandler(logger)
+
+        # Loads Nanopositioners DLL and define arguments and result types for c function
+        self._nanopositionersdll = ctypes.windll.LoadLibrary('SmarActCTL.dll')
+        self._configure_functions()
+
+        #Finds devices connected to controller
+        buffer = ctypes.create_string_buffer(4096)  #the way to have a mutable buffer
+        buffersize = ctypes.c_size_t(ctypes.sizeof(buffer)) #size _t gives c_ulonglong, not as in manual
+        result = self._nanopositionersdll.SA_CTL_FindDevices(None, buffer, buffersize)
+
+        # Handle errors
+        if result:
+            msg_str = 'No MCS2 devices found'
+            self.log.error(msg_str)
+            raise(msg_str)
```

Comment:
A `log.error` automatically raises an exception, thus the `raise` statement can be deleted.